sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/internal/data/MutableDoublePointData.java

```diff
@@ -93,15 +93,15 @@ public boolean equals(Object o) {
     if (this == o) {
       return true;
     }
-    if (o == null || !(o instanceof MutableDoublePointData)) {
+    if (!(o instanceof DoublePointData)) {
       return false;
     }
-    MutableDoublePointData pointData = (MutableDoublePointData) o;
-    return startEpochNanos == pointData.startEpochNanos
-        && epochNanos == pointData.epochNanos
-        && Double.doubleToLongBits(value) == Double.doubleToLongBits(pointData.value)
-        && Objects.equals(attributes, pointData.attributes)
-        && Objects.equals(exemplars, pointData.exemplars);
+    DoublePointData pointData = (DoublePointData) o;
```

Comment:
This generalization applies to all the mutable types. Can we update them all in a separate PR, or is it somehow necessary for this?

Reply:
I update them all - it was only 3 more.